src/lib/api/ext/img-srcset.ts

```diff
@@ -0,0 +1,256 @@
+ *
+ * @see https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture
+ * @see https://www.html5rocks.com/en/tutorials/responsive/picture-element/
+ * @see https://caniuse.com/#search=picture
+ * @see http://scottjehl.github.io/picturefill/
+ */
+@Directive({
+  selector: `
+  img[srcset],
+  img[srcset.xs], img[srcset.sm], img[srcset.md], img[srcset.lg], img[srcset.xl],
+  img[srcset.lt-sm], img[srcset.lt-md], img[srcset.lt-lg], img[srcset.lt-xl],
+  img[srcset.gt-xs], img[srcset.gt-sm], img[srcset.gt-md], img[srcset.gt-lg]
+`
+})
+export class ImgSrcsetDirective extends BaseFxDirective implements OnInit, OnChanges, OnDestroy {
+
+  /* tslint:disable */
+  /**
+   * Intercept srcset assignment so we cache the default static value.
+   * When the responsive breakpoint deactivates,it is possible that fallback static
+   * value (which is used to clear the deactivated value) will be used
+   * (if no other breakpoints activate)
+   */
+  @Input('srcset')        set srcsetBase(val) { this._cacheInput('srcset', val);    }
+
+  @Input('srcset.xs')     set srcsetXs(val)   { this._cacheInput('srcsetXs', val);  }
+  @Input('srcset.sm')     set srcsetSm(val)   { this._cacheInput('srcsetSm', val);  }
+  @Input('srcset.md')     set srcsetMd(val)   { this._cacheInput('srcsetMd', val);  }
+  @Input('srcset.lg')     set srcsetLg(val)   { this._cacheInput('srcsetLg', val);  }
+  @Input('srcset.xl')     set srcsetXl(val)   { this._cacheInput('srcsetXl', val);  }
+
+  @Input('srcset.lt-sm')  set srcsetLtSm(val) { this._cacheInput('srcsetLtSm', val);  }
+  @Input('srcset.lt-md')  set srcsetLtMd(val) { this._cacheInput('srcsetLtMd', val);  }
+  @Input('srcset.lt-lg')  set srcsetLtLg(val) { this._cacheInput('srcsetLtLg', val);  }
+  @Input('srcset.lt-xl')  set srcsetLtXl(val) { this._cacheInput('srcsetLtXl', val);  }
+
+  @Input('srcset.gt-xs')  set srcsetGtXs(val) { this._cacheInput('srcsetGtXs', val);  }
+  @Input('srcset.gt-sm')  set srcsetGtSm(val) { this._cacheInput('srcsetGtSm', val);  }
+  @Input('srcset.gt-md')  set srcsetGtMd(val) { this._cacheInput('srcsetGtMd', val);  }
+  @Input('srcset.gt-lg')  set srcsetGtLg(val) { this._cacheInput('srcsetGtLg', val);  }
+  /* tslint:enable */
+
+  constructor(elRef: ElementRef, renderer: Renderer2, monitor: MediaMonitor) {
+    super(monitor, elRef, renderer);
+  }
+
+  /**
+   * Configure the Picture element with injected <source> elements or support responsive
+   * activation of the standalone- Img element.
+   */
+  ngOnInit() {
+    super.ngOnInit();
+
+    this._configureIsolatedImg();
+    this._configureDefaultSrcset();
+    this._injectSourceElements();
+
+    // Only responsively update srcset values for stand-alone image elements
+    // Fallback to [srcset] value for responsive deactivation
+    this._listenForMediaQueryChanges('src', '', () => {
```

Comment:
I think the key should be 'srcset' rather than 'src' otherwise onMediaQueryChange will never be triggered